motion_computation/src/motion_computation_worker.cpp

```diff
@@ -129,4 +138,218 @@ void MotionComputationWorker::setConfidenceDropRate(double drop_rate)
+
+  if (!local_projector_)
+  {
+    ROS_WARN_STREAM("Motion computation has not received georeference string yet!");
+    return output;
+  }
+
+  // get reference origin in ECEF (convert from cm to m)
+  double ecef_x = (double)msg.trajectory.location.ecef_x/100.0;
+  double ecef_y = (double)msg.trajectory.location.ecef_y/100.0;
+  double ecef_z = (double)msg.trajectory.location.ecef_z/100.0;
+
+  // Convert general information
+  for (size_t i = 0; i < msg.header.sender_bsm_id.size(); i+=2) // convert hex std::string to uint8_t array
+  {
+    int num = 0;
+    sscanf(msg.header.sender_bsm_id.substr(i, i + 2).c_str(), "%x", &num);
+    output.bsm_id.push_back((uint8_t)num);
+  }
+  // first point's timestamp
+  output.header.stamp = ros::Time((double)msg.header.timestamp/ 1000.0);
+
+  // If it is a static object, we finished processing
+  if (msg.trajectory.offsets.size() < 2)
+  {
+    output.dynamic_obj = false;
+    return output;
+  }
+  output.dynamic_obj = true;
+
+  // get planned trajectory points
+  auto prev_pt_msg = msg.trajectory.offsets[0]; // setup first point to be processed later
+  cav_msgs::PredictedState prev_state;
+  lanelet::BasicPoint3d prev_pt_ecef {ecef_x + (double)prev_pt_msg.offset_x /100.0, ecef_y + (double)prev_pt_msg.offset_y /100.0, ecef_z + (double)prev_pt_msg.offset_z /100.0};
+
+  auto prev_pt_map = local_projector_->projectECEF(prev_pt_ecef, -1);
+
+  for (size_t i = 1; i < msg.trajectory.offsets.size(); i ++)
+  {
+    auto curr_pt_msg = msg.trajectory.offsets[i];
+    lanelet::BasicPoint3d curr_pt_ecef {ecef_x + (double)curr_pt_msg.offset_x /100.0, ecef_y + (double)curr_pt_msg.offset_y /100.0, ecef_z + (double)curr_pt_msg.offset_z /100.0};
+    auto curr_pt_map = local_projector_->projectECEF(curr_pt_ecef, -1);
+
+    cav_msgs::PredictedState curr_state;
+
+    if (i == 1) // First point's state should be stored outside "predictions"
+    {
+      curr_state = composePredictedState(curr_pt_map, prev_pt_map, output.header.stamp); 
+      output.pose.pose = curr_state.predicted_position;
+      output.velocity.twist = curr_state.predicted_velocity;
+    }
+    else    
+    {
+      curr_state = composePredictedState(curr_pt_map, prev_pt_map, prev_state.header.stamp + ros::Duration(mobility_path_prediction_time_step_));
+      output.predictions.push_back(curr_state);
+    }
+
+    if (i == msg.trajectory.offsets.size() - 1) // if last point, copy the prev_state velocity & orientation to the last point too
+    {
+      curr_state = composePredictedState(curr_pt_map, prev_pt_map, curr_state.header.stamp + ros::Duration(mobility_path_prediction_time_step_));
+      curr_state.predicted_position.position.x = curr_pt_map.x();
+      curr_state.predicted_position.position.y = curr_pt_map.y();
+      curr_state.predicted_position.position.z = curr_pt_map.z();
+      output.predictions.push_back(curr_state);
+    }
+    
+    prev_state = curr_state;
+    prev_pt_map = curr_pt_map;
+  }
+
+  return output;
```

Comment:
It looks to me like we are reaching this point without ever setting the current state for the external object. Only the positions seem to be set. Is that being done somewhere I am not seeing?

Reply:
if (i==1) is handling the current state of the object. Pose includes position and orientation. Velocity is set there as well. Let me know if I am missing anything

Reply:
Looks like you have the pose and velocity set. What about the object type, precense vector, and id?

Reply:
Added those as well